src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/TypeHelpers.cs

```diff
@@ -606,13 +606,54 @@ internal static bool TryGetDebuggerDisplayInfo(this DkmClrValue value, out Debug
             return displayInfo.HasValues;
         }
 
+        /// <summary>
+        /// Custom comparer for the GetDebuggerCustomUIVisualizerInfo method so that our custom set can
+        /// eliminate duplicated visualizer attributes from a type due to inheritance.
+        /// </summary>
+        private class DkmClrDebuggerVisualizerAttributeComparer : IEqualityComparer<DkmClrDebuggerVisualizerAttribute>
+        {
+            public bool Equals(DkmClrDebuggerVisualizerAttribute x, DkmClrDebuggerVisualizerAttribute y)
+            {
+                if (ReferenceEquals(x, y))
+                {
+                    return true;
+                }
+
+                return x.VisualizerDescription == y.VisualizerDescription &&
+                       x.UISideVisualizerTypeName == y.UISideVisualizerTypeName &&
+                       x.UISideVisualizerAssemblyName == y.UISideVisualizerAssemblyName &&
+                       x.UISideVisualizerAssemblyLocation == y.UISideVisualizerAssemblyLocation &&
+                       x.DebuggeeSideVisualizerTypeName == y.DebuggeeSideVisualizerTypeName &&
+                       x.DebuggeeSideVisualizerAssemblyName == y.DebuggeeSideVisualizerAssemblyName;
+            }
+
+            public int GetHashCode(DkmClrDebuggerVisualizerAttribute obj)
+            {
+                const int prime = 31;
+
+                int hash = 1;
+                if (obj != null)
+                {
+                    hash = (hash * prime) + obj.VisualizerDescription.GetHashCode();
```

Comment:
>VisualizerDescription [](http://example.com/codeflow?start=48&length=21)

Are all of these string fields guaranteed to be non-null? And can we simplify `GetHashCode()`? For instance, consider simply returning the hash code of the most significant field.
```
return obj?.UISideVisualizerTypeName?.GetHashCode() ?? 0;
```